doc/api/image_api.rst

```diff
@@ -3,6 +3,172 @@
+   :parts: 1
+   :private-bases:
+
+
+.. autosummary::
+   :toctree: _as_gen/
+   :template: autosummary.rst
+
+   _ImageBase
+   AxesImage
+   NonUniformImage
+   PcolorImage
+   FigureImage
+   BboxImage
+
+
+Resampling
+~~~~~~~~~~
+
+When Matplotlib rasterizes an image when saving / displaying a Figure
+we need to, in general, resample the data (either up or down) in
+addition to normalizing and color mapping it.  This is because the
+exact size of the input, in "data" pixels, will not match the size, in
+"screen" pixels, of the output.  The details of how we do the
+resampling is controlled by the *interpolation* specified.  This
+resampling process can introduce a variety of artifacts and the
+default interpolation is chosen to avoid aliasis in common cases (see
+:doc:`/gallery/images_contours_and_fields/image_antialiasing`).
+
+Colormapping
+~~~~~~~~~~~~
+
+The processing steps for rendering a pseudo color image are:
+
+1. rasample to user input to the required dimensions
+2. normalize the user data via a `~.colors.Normalize` instance
+3. colormap from the normalized data to RGBA via a `~.colors.Colormap` instance
+
+Prior to Matplotlib 2.0 we did the normalization and colormapping
+first and then resampled to fit the screen.  However this can produce
+artifacts in the visualization when the data is changing close to the
+full range on the scale of a few pixels with most colormaps due to the
+interpolation in RGB space producing colors that are not in the
+colormap.
+
+
+Floating point and you
+~~~~~~~~~~~~~~~~~~~~~~
+
+Floating point numbers, despite being ubiquitous, are not fully
+understood by most practitioners.  For a through review of how
+floating point numbers work see `Goldberg, ACM Computing
+Surveys (1991) 10.1145/103162.103163
+<https://doi.org/10.1145/103162.103163>`__ or the `IEEE Standard for
+Floating Point Arithmetic ( IEEE std 754) 10.1109/IEEESTD.2008.4610935
+<https://doi.org/10.1109/IEEESTD.2008.4610935>`__ (both behind
+paywalls).  For the purposes of this discussion we need to know:
+
+1. There are only a finite number "floating point numbers" (that is
+   values that can be represented by a IEEE float in the computer) and
+   hence can not exactly represent all Real Numbers.  Between those
+   two Real Numbers there is an infinite number of Real numbers, hence
+   the floating point numbers and computation expressed in a computer
+   are an approximation of the Real Numbers.
+2. The absolute distance between adjacent floating point numbers
+   scales with the magnitude, while the relative distance remains
+   the same.  This is a consequence of the implementation of IEEE
+   floats.
+3. During computation results are rounded to the nearest
+   represent-able value.  Working with numbers that are either almost
+   identical or vastly different orders of magnitude exasperates the
```

Comment:
"exasperates" → "exaggerates"? The former implies that the numbers lose their patience.